presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -419,43 +411,47 @@ private Table createTable(
         ImmutableList.Builder<FieldSchema> columns = ImmutableList.builder();
         for (HiveColumnHandle columnHandle : columnHandles) {
             String name = columnHandle.getName();
-            String type = columnHandle.getHiveType().getHiveTypeName();
+            HiveType type = columnHandle.getHiveType();
             if (name.equals(SAMPLE_WEIGHT_COLUMN_NAME)) {
-                columns.add(new FieldSchema(name, type, "Presto sample weight column"));
+                columns.add(new FieldSchema(name, type, Optional.of("Presto sample weight column")));
                 sampled = true;
             }
             else if (!partitionColumnNames.contains(name)) {
                 verify(!columnHandle.isPartitionKey(), "Column handles are not consistent with partitioned by property");
-                columns.add(new FieldSchema(name, type, null));
+                columns.add(new FieldSchema(name, type, Optional.empty()));
             }
             else {
                 verify(columnHandle.isPartitionKey(), "Column handles are not consistent with partitioned by property");
             }
         }
 
-        Table table = new Table();
-        table.setDbName(schemaName);
-        table.setTableName(tableName);
-        table.setOwner(tableOwner);
-        table.setTableType(TableType.MANAGED_TABLE.toString());
         String tableComment = "Created by Presto";
         if (sampled) {
             tableComment = "Sampled table created by Presto. Only query this table from Hive if you understand how Presto implements sampling.";
         }
-        table.setParameters(ImmutableMap.<String, String>builder()
-                .put("comment", tableComment)
-                .putAll(additionalTableParameters)
-                .build());
-        table.setPartitionKeys(partitionColumns);
-        table.setSd(makeStorageDescriptor(tableName, hiveStorageFormat, targetPath, columns.build(), bucketProperty));
-
         PrivilegeGrantInfo allPrivileges = new PrivilegeGrantInfo("all", 0, tableOwner, PrincipalType.USER, true);
-        table.setPrivileges(new PrincipalPrivilegeSet(
+        Table.Builder tableBuilder = new Table.Builder()
+                .setDatabaseName(schemaName)
+                .setTableName(tableName)
+                .setOwner(tableOwner)
+                .setTableType(TableType.MANAGED_TABLE.name())
+                .setDataColumns(columns.build())
+                .setPartitionColumns(partitionColumns)
+                .setParameters(ImmutableMap.<String, String>builder()
+                        .put("comment", tableComment)
+                        .putAll(additionalTableParameters)
+                        .build());
+        tableBuilder.getStorageBuilder()
+                .setStorageFormat(StorageFormat.fromHiveStorageFormat(hiveStorageFormat))
```

Comment:
I can do that. Given `Table`/`Partition`/`Storage` and their corresponding `Builder`s are just entity classes, I thought it would be a bad idea to give them knowledge about things like `HiveStorageFormat`, which is the `enum`s that Presto supports writing to.